plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByTitleProvider.java

```diff
@@ -0,0 +1,186 @@
+		Collection<Viewpoint> selectedViewpoints = session.getSelectedViewpoints(false);
+
+		for (DRepresentation representation : representations) {
+			if (representationName.equals(representation.getName()) && representation.eContainer() instanceof DView) {
+				DView dView = (DView) representation.eContainer();
+				Viewpoint vp = dView.getViewpoint();
+				if (selectedViewpoints.contains(vp)) {
+					return representation;
+				}
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Get the GmF Diagram instance corresponding to a viewpoint
+	 * DSemanticDiagram
+	 * 
+	 * @param semanticDiagram
+	 * @return
+	 */
+	private Diagram getGmfDiagram(DSemanticDiagram semanticDiagram) {
+		for (final AnnotationEntry annotation : new DDiagramQuery(semanticDiagram)
+				.getAnnotation(CustomDataConstants.GMF_DIAGRAMS)) {
+			EObject eObject = annotation.getData();
+			if (eObject instanceof Diagram) {
+				final Diagram diagramInResource = (Diagram) eObject;
+				final EObject semanticElement = ViewUtil.resolveSemanticElement(diagramInResource);
+				if (semanticElement.equals(semanticDiagram)) {
+					return diagramInResource;
+				}
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public List<String> getRepresentationImagePath(Map<String, Object> parameters) throws ProviderException {
+
+		EObject rootObject = (EObject) parameters.get(ProviderConstants.KEY_CONF_ROOT_OBJECT);
+		String rootPath = (String) parameters.get(ProviderConstants.KEY_PROJECT_ROOT_PATH);
+		Session session = SessionManager.INSTANCE.getSession(rootObject);
+		Object representationName = parameters.get(KEY_REPRESENTATION_TITLE);
+		if (!(representationName instanceof String)) {
+			throw new ProviderException("Image cannot be computed because no representation title has been provided to the provider \""+this.getClass().getName()+"\"");
+		} else {
+			DRepresentation representation = getAssociatedRepresentationByName((String) representationName,session);
+			if (representation instanceof DSemanticDiagram) {
+				DSemanticDiagram dsd = (DSemanticDiagram) representation;
+				String filePath = getDiagramImageFilename(dsd,rootPath);
+				File file = new File(filePath);
+				file.getParentFile().mkdirs();
+				final IPath path = new Path(filePath);
+				final Diagram gmfDiagram = getGmfDiagram(dsd);
+
+				// Use the DiagramEditPartService to use the figure validation
+				// infinite loop safe ViewpointDiagramGraphicalViewer.
+				final CopyToImageUtil imageUtility = new DiagramEditPartService();
+
+				final EditingDomain editingDomain = session.getTransactionalEditingDomain();
+				final Diagram realOne = (Diagram) editingDomain.getResourceSet()
+						.getEObject(EcoreUtil.getURI(gmfDiagram), true);
+				try {
+					imageUtility.copyToImage(realOne, path, ImageFileFormat.JPEG, new NullProgressMonitor(),
+							PreferencesHint.USE_DEFAULTS);
+					List<String> resultList = new ArrayList<String>(1);
+					resultList.add(filePath);
+					return resultList;
```

Comment:
Same comment as in the other provider : this should be put in a separate method. 
Plus, it can be factorized between the two providers.